jsonnet/components/thanos-ruler.libsonnet

```diff
@@ -1,495 +1,485 @@
-      },
-    },
-    spec: {
-      selector: {
-        matchLabels: cfg.labels,
+    service+: {
+      metadata+: {
+        annotations: {
+          'service.beta.openshift.io/serving-cert-secret-name': 'thanos-ruler-tls',
+        },
       },
-      endpoints: [
-        {
-          port: 'web',
-          interval: '30s',
-          scheme: 'https',
-          tlsConfig: {
-            caFile: '/etc/prometheus/configmaps/serving-certs-ca-bundle/service-ca.crt',
-            serverName: 'server-name-replaced-at-runtime',
-            certFile: '/etc/prometheus/secrets/metrics-client-certs/tls.crt',
-            keyFile: '/etc/prometheus/secrets/metrics-client-certs/tls.key',
-          },
-          bearerTokenFile: '/var/run/secrets/kubernetes.io/serviceaccount/token',
+      spec+: {
+        ports: [{
+          name: 'web',
+          port: 9091,
+          targetPort: 'web',
+        }, {
+          name: 'grpc',
+          port: 10901,
+          targetPort: 'grpc',
+        }],
+        selector: {
+          'app.kubernetes.io/name': tr.config.name,
+          'thanos-ruler': 'user-workload',
         },
-      ],
-    },
-  },
-
-  thanosRuler: {
-    apiVersion: 'monitoring.coreos.com/v1',
-    kind: 'ThanosRuler',
-    metadata: {
-      name: cfg.name,
-      namespace: cfg.namespace,
-      labels: {
-        thanosRulerName: cfg.name,
+        sessionAffinity: 'ClientIP',
+        type: 'ClusterIP',
+        clusterIP:: {},
       },
     },
-    spec: {
-      affinity: {
-        podAntiAffinity: {
-          requiredDuringSchedulingIgnoredDuringExecution: [{
-            labelSelector: {
-              matchLabels: cfg.selectorLabels,
+
+    serviceMonitor+:
+      {
+        spec+: {
+          endpoints: [
+            {
+              port: 'web',
+              interval: '30s',
+              scheme: 'https',
+              tlsConfig: {
+                caFile: '/etc/prometheus/configmaps/serving-certs-ca-bundle/service-ca.crt',
+                serverName: 'server-name-replaced-at-runtime',
+                certFile: '/etc/prometheus/secrets/metrics-client-certs/tls.crt',
+                keyFile: '/etc/prometheus/secrets/metrics-client-certs/tls.key',
+              },
+              bearerTokenFile: '/var/run/secrets/kubernetes.io/serviceaccount/token',
             },
-            namespaces: [cfg.namespace],
-            topologyKey: 'kubernetes.io/hostname',
-          }],
+          ],
         },
       },
-      securityContext: {
-        fsGroup: 65534,
-        runAsNonRoot: true,
-        runAsUser: 65534,
-      },
-      replicas: cfg.replicas,
-      resources: {
-        requests: {
-          memory: '21Mi',
-          cpu: '1m',
+
+
+    thanosRuler: {
+      apiVersion: 'monitoring.coreos.com/v1',
+      kind: 'ThanosRuler',
+      metadata: {
+        name: cfg.crName,
+        namespace: tr.config.namespace,
+        labels: {
+          thanosRulerName: cfg.crName,
```

Comment:
this is definitely not introduced by your PR but I wonder why we have this label on the ThanosRuler resource? :thinking:

Reply:
I'm not sure. I can remove it if we like. I was just trying to keep the diff in this PR to a minimum.

Reply:
We should try to remove it and see what the CI says but not in this PR.